src/fastgltf.cpp

```diff
@@ -4524,6 +4560,29 @@ void fg::Exporter::writeImages(const Asset& asset, std::string& json) {
 		json += ',';
 
 	json += "\"images\":[";
+
+	auto imagePathsHasPath = [&] (const std::filesystem::path &path) {
+		for (auto& imagePath : imagePaths) {
+			if (imagePath.has_value() && *imagePath == path) {
+				return true;
+			}
+		}
+		return false;
+	};
+
+	auto addPath = [&] (const std::filesystem::path &path) {
+		// while it's still present, add "-1", "-2", "-3", etc. to the path
+		// e.g. RoughnessGrid.png -> RoughnessGrid-1.png
+		auto newPath = path;
+		int i = 1;
+		while (imagePathsHasPath(newPath)) {
+			newPath = path.parent_path() / (path.stem().string() + "-" + std::to_string(i) + path.extension().string());
+			++i;
+		}
+		imagePaths.emplace_back(newPath);
+		return newPath;
+	};
```

Comment:
Why are these lambdas? Couldn't they just be part of the `getImageFilePath` function?